xarray/tests/conftest.py

```diff
@@ -191,7 +191,7 @@ def create_test_datatree():
     """
 
     def _create_test_datatree(modify=lambda ds: ds):
-        set1_data = modify(xr.Dataset({"a": 0, "b": 1}))
+        set1_data = modify(xr.Dataset({"a": 1, "b": 2}))
```

Comment:
can this be safely reverted?

Reply:
Reverting this breaks the `test_to_zarr_compute_false` test. I think that's because zarr-python sees the array with a value `0` matching its default `fill_value`, and chooses not to write this chunk as an optimization. But the test expects all (non-dask) arrays to become chunks on disk and so fails.

However even the new values used by @aladinor have problems - they trigger this bug https://github.com/pydata/xarray/issues/10147. I haven't been able to fully get around these bugs, but none of this is a datatree-specific issue though, so I think it is actually okay to change the test fixture...

Reply:
I was able to fully revert this once I taught the test assertion to understand the expected zarr behaviour for scalars and for arrays containgin only `fill_value`.

https://github.com/aladinor/xarray/pull/1/commits/0969422b5d4d302a0bdecf1cdf85dcfd642e4005